app/angular/components/queryExpressionModal/index.js

```diff
@@ -0,0 +1,145 @@
+import angular from "angular";
+import template from "./template.html";
+import "./styles.scss";
+
+const app = angular.module("app.queryExpressionModalController", []);
+
+const AND_OPERATOR = 'AND';
+const OR_OPERATOR = 'OR';
+
+const Controller = function ($filter) {
+	const $ctrl = this;
+	$ctrl.submitted = false;
+	$ctrl.conditions = [];
+	$ctrl.joins = [];
+	$ctrl.tableColumns = [];
+
+	const defaultJoin = {
+		columnName: null,
+		columnName2: null,
```

Comment:
Instead of using a number, wouldn't it be better to name it as `origin` and `target`?

```suggestion
		columnNameOrigin: null,
		columnNameTarget: null,
```